src/main/java/com/list/todo/configurations/SecurityConfiguration.java

```diff
@@ -1,61 +1,105 @@
 package com.list.todo.configurations;
 
+import com.list.todo.security.CustomBasicAuthenticationEntryPoint;
+import com.list.todo.security.JwtAuthenticationEntryPoint;
+import com.list.todo.security.JwtAuthenticationFilter;
 import com.list.todo.services.UserService;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.http.HttpMethod;
+import org.springframework.security.authentication.AuthenticationManager;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.builders.WebSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
 import org.springframework.security.config.http.SessionCreationPolicy;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.security.config.BeanIds;
+
 
 @Configuration
 @EnableWebSecurity
+@EnableGlobalMethodSecurity(
+        securedEnabled = true,
+        jsr250Enabled = true,
+        prePostEnabled = true
+)
 public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
 
     private final UserService userService;
+	private JwtAuthenticationEntryPoint unauthorizedHandler;
 
     @Autowired
-    public SecurityConfiguration(UserService userService) {
-        this.userService = userService;
+    public SecurityConfiguration(UserService userService,
+			JwtAuthenticationEntryPoint unauthorizedHandler) {
+		this.userService = userService;
+		this.unauthorizedHandler = unauthorizedHandler;
+	}
+
+	@Bean
+    public JwtAuthenticationFilter jwtAuthenticationFilter() {
```

Comment:
please fix indentation

Reply:
In IDE we don't have this indentation :)